client/src/utils/gettext.test.js

```diff
@@ -9,68 +9,74 @@ import {
 const STRING =
   'The name wagtail stems from the constant sideways wagging of the tail.';
 
+afterEach(() => {
+  // clear any mocked globals after each test
+  if (window.django) {
+    delete window.django;
+  }
+});
+
 describe('gettext', () => {
-  beforeEach(() => {
-    window.django = { gettext: jest.fn((val) => val) };
+  it('should return the provided string if Django gettext is not loaded', () => {
+    expect(gettext(STRING)).toEqual(STRING);
   });
 
-  it('should return the value based on the django util', () => {
-    expect(window.django.gettext).not.toHaveBeenCalled();
-    expect(gettext(STRING)).toEqual(STRING);
-    expect(window.django.gettext).toHaveBeenCalledWith(STRING);
+  it('should call the global Django util if loaded', () => {
+    const gettextMock = jest.fn();
+    window.django = { gettext: gettextMock };
+    gettext(STRING);
+    expect(gettextMock).toHaveBeenCalledWith(STRING);
   });
 });
 
 describe('ngettext', () => {
-  beforeEach(() => {
-    window.django = { ngettext: jest.fn((val) => val) };
+  it('should return the first param if Django ngettext is not loaded', () => {
+    expect(ngettext('One bird', 'Many birds', 2)).toEqual('One bird');
```

Comment:
@lb- Why doesn't this return "Many birds"?

Reply:
Because. This is the default behaviour if the global Django util is not loaded. 

That's been the default behaviour from the time that this code was introduced. 

Sorry if it's not clear but this unit test checks that the code has some default behaviour (not an error) if the global is loaded. 


The unit test after checks that if the global is loaded, it gets called. 

Would it help if I just changed the number to 1?

Reply:
Ah yep, makes sense, thanks! I missed the updated description of the test.

Just so I'm clear - the Django util functions being absent (and the use of these fallback functions) isn't something we expect to happen in "real-world" code, but only for the purposes of unit tests, right? I'd be a bit happier if the fallback implementation became `return (count == 1 ? singular : plural)`, so that the behaviour in tests is equivalent to running it for real in an English-language locale.

Reply:
Sounds good. I'll prepare another PR. 

Alternatively we could make it error loudly if the global is not there, it's a bit more work to change it but maybe that is more 'correct'?

We would however need to ensure that all unit tests have a mock of the global loaded (can be done in jest.setup) or a mock for each of the utils.

It's just a bit tricky either way as the Django import adds a global instead of being an npm module import. So we kind of need to think about the low chance of it not working but also handle the import global not existing sometimes like unit tests or maybe Storybook in edge cars.

Reply:
I think the "make it error loudly" approach would probably just make life difficult for us :-) We're best served by having a simple fallback implementation that doesn't have any external dependencies, but is close enough to the "real" behaviour that our tests don't have to be contrived to work around the differences (e.g. being able to assert that the output contains "2 Errors" rather than "2 Error").